File: python/exercise_101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    def isSymmetrical(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if p == None and q == None:
            return True
        if p == None or q == None:
            return False
        print([p.val,q.val])
        return self.isSameTree(p.left,q.right) and (p.val == q.val) and self.isSameTree(p.right,q.left)
    
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if root.left == None and root.right == None:
            return True
        if root.left == None or root.right == None:
            return False
        return self.isSymmetrical(root.left,root.right)

File: java/tree/exercise_101.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if (root == null) return true;
        return isSymmetrical(root.left,root.right);
    }
    public boolean isSymmetrical(TreeNode p, TreeNode q) {
        if(p == null && q != null){
            return false;
        }
        if(q == null && p != null){
            return false;
        }
        if(p == null && q == null){
            return true;
        }
        return p.val == q.val && isSymmetrical(p.left,q.right) && isSymmetrical(p.right,q.left);
    }
}